dexlib2/src/main/java/org/jf/dexlib2/dexbacked/util/DebugInfo.java

```diff
@@ -270,10 +275,13 @@ protected DebugItem readNextItem(@Nonnull DexReader reader) {
                                 int adjusted = next - 0x0A;
                                 codeAddress += adjusted / 15;
                                 lineNumber += (adjusted % 15) - 4;
-                                return new ImmutableLineNumber(codeAddress, lineNumber);
+                                return codeAddress <= lastInstructionAddress
```

Comment:
Isn't this condition guaranteed by the loop conditonal?

Reply:
The default event may adjust the code address on line 276. If the resulting pc is past the last instructions start address then the present event should not be included in the events for this method. This does mean that we may need to iterate over one more event than is actually part of the events for the method.

Reply:
Are there any other comments regarding this change, or is in shape for landing?

Reply:
> Are there any other comments regarding this change, or is in shape for landing?

no progress

Reply:
sorry about the rather long delay. I think it's become clear to me that I no longer have the motivation to maintain smali, and should probably try to find a new maintainer. If you would like to take over, or know someone at Google or otherwise that would, feel free to email me or whatever and we can discuss.